bundles/org.openhab.binding.gree/src/main/java/org/openhab/binding/gree/internal/handler/GreeHandler.java

```diff
@@ -138,62 +140,78 @@ public void handleCommand(ChannelUID channelUID, Command command) {
-                        break;
-                    case LIGHT_CHANNEL:
-                        device.setDeviceLight(socket, getOnOff(command));
-                        break;
-                    case TARGET_TEMP_CHANNEL:
-                        // Set value, read back effective one and update channel
-                        // e.g. 22.5C will result in 22.0, because the AC doesn't support half-steps for C
-                        device.setDeviceTempSet(socket, convertTemp(command));
-                        break;
-                    case SWINGUD_CHANNEL:
-                        device.setDeviceSwingUpDown(socket, getNumber(command));
-                        break;
-                    case SWINGLR_CHANNEL:
-                        device.setDeviceSwingLeftRight(socket, getNumber(command));
-                        break;
-                    case WINDSPEED_CHANNEL:
-                        device.setDeviceWindspeed(socket, getNumber(command));
-                        break;
-                    case QUIET_CHANNEL:
-                        handleQuietCommand(socket, command);
-                        break;
-                    case AIR_CHANNEL:
-                        device.setDeviceAir(socket, getOnOff(command));
-                        break;
-                    case DRY_CHANNEL:
-                        device.setDeviceDry(socket, getOnOff(command));
-                        break;
-                    case HEALTH_CHANNEL:
-                        device.setDeviceHealth(socket, getOnOff(command));
-                        break;
-                    case PWRSAV_CHANNEL:
-                        device.setDevicePwrSaving(socket, getOnOff(command));
-                        break;
+
+            boolean retry = false;
+            do {
+                try {
+                    retry = false;
+                    sendRequest(channelId, command);
+                    // force refresh on next status refresh cycle
+                    forceRefresh = true;
+                    apiRetries = 0;
+                } catch (IllegalArgumentException e) {
+                    logInfo("command.invarg", command, channelId);
+                } catch (GreeException e) {
+                    apiRetries++;
+                    if (apiRetries >= MAX_API_RETRIES) {
+                        String message = logInfo(
+                                messages.get("command.exception", command, channelId) + ": " + e.getMessage());
+                        updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR, message);
+                    } else {
+                        retry = true;
+                        logger.debug("{}: Command {} failed for channel {}, retry", thingId, command, channelId);
+                    }
+                } catch (RuntimeException e) {
+                    logger.warn("{}: {}", thingId, messages.get("command.exception", command, channelId), e);
                 }
+            } while (retry);
```

Comment:
Does it make sense to retry 3 times in a row without some wait time?

Reply:
There is no delay in submitting commands: 3 retries in a row
only the status poll has delays, which means "unit was not available over the last 3 poll cycles"

I think that makes sense. Nevertheless I changed the code to use a local variable rather than the instance variable used to count the status poll retries.